Update ADR-0001-egress-control.md (#2211)
cloud-gov/product: adr/ADR-0001-egress-control.md

- After we have ASGs and reference implementations in place, we could conceivably
  make a broker that creates/manages proxies for users
- This solution probably only solves access to HTTP(S)

## Addenda

### s3 egress in restricted ASG

> To simplify client accessibility for S3 we have added s3 access in the restricted ASG.

Reviewer: ```suggestion

To simplify client accessibility for S3 we have added s3 access in the restricted ASG.
```

- We accomplished this by leveraging a private endpoint for s3 in each VPC and setting a policy that allows access to buckets by users belonging to a list of accounts (ie the AWS accounts owned by cloud.gov)

>   - cg-provision contains the policy creation and outputs the ip ranges and vpe dns names
>   - cg-deploy-cf contains changes to the ASGs to include the ip ranges and dns names output by cg-provision
>   - the VPCs in AWS each contain a pair of s3 private endpoints ( gateway and singular) for access to s3 over the AWS internal network. 

Reviewer: ```suggestion
  - `cg-provision` contains the policy creation and outputs the IP ranges and VPC dns names
  - `cg-deploy-cf` contains changes to the ASGs to include the IP ranges and DNS names output by cg-provision
  - the VPCs in AWS each contain a pair of S3 private endpoints ( gateway and singular) for access to S3 over the AWS internal network. 
```

Nitpicking: acronyms should be capitalized. 

It would also be helpful for things that are repo names to link to the repos, for easier reference

- Clients wanting to access their own buckets not provisioned by service broker, must still use public ASG or a proxy to a public ASG. 

>   - those buckets that exsist in other regions from the cloud.gov current production region ( eg. us-gov-west-1 ) can be accessed transparently with standard api endpoints

Reviewer: ```suggestion
  - those buckets that exist in other regions from the cloud.gov current production region ( `us-gov-west-1` ) can be accessed transparently with standard API endpoints
```

>   - those buckets that exsist in the same region as cloud.gov production will need to use an alternate endpoint that is created by our Terraform: `*.vpce-01beaa66570dfb2b9-1hlav4x8.s3.us-gov-west-1.vpce.amazonaws.com`. This is due to the fact that private endpoints are regional in AWS and hence by default all traffic to s3 buckets in `us-gov-west-1` will have the implemented policy applied. 

Reviewer: ```suggestion
  - buckets that exist in the same region as cloud.gov production will need to use an alternate endpoint that is created by our Terraform: `*.vpce-01beaa66570dfb2b9-1hlav4x8.s3.us-gov-west-1.vpce.amazonaws.com`. This is due to the fact that private endpoints are regional in AWS and hence by default all traffic to S3 buckets in `us-gov-west-1` will have the implemented policy applied. 
```

- After we have ASGs and reference implementations in place, we could conceivably
  make a broker that creates/manages proxies for users
- This solution probably only solves access to HTTP(S)

## Addenda

### s3 egress in restricted ASG

To simplify client accessibility for S3 we have added s3 access in the restricted ASG.
- We accomplished this by leveraging a private endpoint for s3 in each VPC and setting a policy that allows access to buckets by users belonging to a list of accounts (ie the AWS accounts owned by cloud.gov)
  - `cg-provision` contains the policy creation and outputs the IP ranges and VPC dns names

>   - `cg-deploy-cf` contains changes to the ASGs to include the IP ranges and DNS names output by cg-provision

Author: ```suggestion
  - [cg-deploy-cf](https://github.com/cloud-gov/cg-deploy-cf) contains changes to the ASGs to include the IP ranges and DNS names output by cg-provision
```
